### PR TITLE
[CPM-84] Course Regulations UI 

### PR DESCRIPTION
**Pull request by Richardng12**
- Regulations Modal added
- Regulations Table added
- Connecting to back-end will be done in CPM-85
- Regulations are deletable

Do we need to add a confirmation alert dialog for deleting a regulation?
Please ignore files that arent in coursesPage (rebasing)

### CODE REVIEW

File: client/src/pages/CoursesPage/RegulationTable.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                </Button>
+                {allRegsArray.map((reg) => (
+                    <Flex p={1} shadow="md" borderWidth="1px" flex="1" rounded="md" justify="space-between" direction="row">
+                        <Text>
+                            {reg.map((value, i) => (
+                                <>{i !== reg.length - 1 ? `${value} or ` : ` ${value}`}</>
+                            ))}
+                        </Text>
+                        {/* <Text>{reg}</Text> */}
+                        {/* <Text>{i != allRegsArray.length - 1 ? " or " : ""}</Text> */}
```
Comment (FocalChord) [inline client/src/pages/CoursesPage/RegulationTable.js:37]: Can we remove these comments

File: client/src/pages/CoursesPage/useCourses.js [hunk 2]
```diff
@@ -24,14 +24,16 @@ const useCourses = () => {
...
+            .then((res) => setCourses(res))
+            .catch((e) => console.log(e));
+    };
 
     const updateCourse = (course) => {
-        MomentumClient.updateCourse(course).then(r => fetchAllCourses())
-            .catch(e => console.log(e));
-    }
+        MomentumClient.updateCourse(course)
+            .then((r) => fetchAllCourses())
```
Comment (FocalChord) [inline client/src/pages/CoursesPage/useCourses.js:34]: Redundant `r`

**Review by daltonlim — APPROVED** [review]
LGTM

**Review by FocalChord — APPROVED** [review]
good work :shipit:

**Pull request merged**